postproengine/wake_meandering.py

```diff
@@ -0,0 +1,365 @@
+# Get the location where this script is being run
+import sys, os
+scriptpath = os.path.dirname(os.path.realpath(__file__))
+basepath   = os.path.dirname(scriptpath)
+# Add any possible locations of amr-wind-frontend here
+amrwindfedirs = ['../',
+                 basepath]
+for x in amrwindfedirs: sys.path.insert(1, x)
+
+from postproengine import registerplugin, mergedicts, registeraction
+import postproamrwindsample_xarray as ppsamplexr
+import postproamrwindsample as ppsample
+import numpy as np
+import pandas as pd
+import os
+import pickle
+import matplotlib.pyplot as plt
+from mpl_toolkits.axes_grid1 import make_axes_locatable
+# samwichdir = '/projects/wind_uq/gyalla/src/waketracking/'
+# sys.path.insert(0,samwichdir)
+try:
+    from samwich.dataloaders import PlanarData
+    from samwich.waketrackers import track
+    usesamwhich=True
+except:
+    usesamwhich=False
+
+"""
+Plugin for computing wake meandering statistics
+
+See README.md for details on the structure of classes here
+"""
+
+def get_wake_centers(u,YY,ZZ,yhub,method='ConstantArea',weighting=lambda u: np.ones_like(u),args=None):
+    datadict = {}
+    datadict['y'] = np.copy(YY[:,:].T)
+    datadict['z'] = np.copy(ZZ[:,:].T)
+    datadict['u'] = np.array([np.copy(u[i,:,:,0].T) for i in range(0,u.shape[0])])
+    wakedata = PlanarData(datadict)
+    wake = track(wakedata.sliceI(),method=method,verbose=True)
+    wake.remove_shear(method='fringe',Navg=u.shape[0])
+    wake.wake_tracked = False
+    if method=='Gaussian':
+        #max_VD = -np.min(wake.u,axis=(1,2))  # max velocity deficit, u.shape == (Ntimes,Nh,Nv)
+        #sigma_opt = np.sqrt(ref_thrust / (np.pi*max_VD*(2*Uref - max_VD)))
+        yc,zc = wake.find_centers(umin=None,sigma=args)
+    if method=='ConstantArea':
+        yc,zc = wake.find_centers(args,weighted_center=weighting)
+    if method=='ConstantFlux':
+        flux = lambda u,u_w: -u * u_w  # function arguments correspond to field_names
+        yc,zc = wake.find_centers(args,flux_function=flux,field_names=('u','u_tot'))
+    return wake, yc + yhub ,zc
```

Comment:
`yc + yhub` -> does `wake.find_centers` return `yc` assuming the origin is the center of the plane?

Reply:
The samwich package seems to zero the center of the y-grid. The hub-height location of y is therefore passed as an argument to get_wake_centers() so that yc + yhub is returned as the wake center. zc does not need this adjustment

Reply:
is yhub the center of the y-grid always?

Reply:
Good catch, samwich centers the results on the mid-point of the y-grid, not necessarily yhub. So I have added the grid mid point back to the wake centers to transform to the original coordinates. yhub and zhub are now optional parameters which are only needed for computing the mean and std of the distance to the hub